bmq-sdk/src/main/java/com/bloomberg/bmq/impl/infr/net/NettyTcpConnection.java

```diff
@@ -745,19 +742,13 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
 
             ByteBuf byteBuf = (ByteBuf) msg;
             try {
-                // Make a copy of bytes present in 'byteBuf'.  This is needed
-                // so that the lifetime of 'readBuffer' (and ByteBuffer[]
-                // obtained from it via 'readBuffer.reset()') can be
-                // decoupled with the associated 'byteBuf'.  As per netty docs,
-                // a ByteBuf object is refcounted, and user must invoke
-                // 'ByteBuf.release' to decrement the counter when done.  Since
-                // there is no place in the logic where we can confidentally
-                // invoke 'ByteBuf.release' (BlazingMQ user may keep a PUSH message
-                // for ever), we make a copy of bytes present in 'ByteBuf' and
-                // invoke 'ByteBuf.release' right away.  This copying is
-                // unfortunate, and we will investigate ways to avoid this
-                // copy later.
-                readBuffer.writeBytes(byteBuf.nioBuffer());
+                // This still makes a copy of the underlying bytes, but let netty do it
+                // instead of doing it ourselves.
+                //
+                // TODO - can this copy be eliminated?
+                ByteBuffer nioBuf = ByteBuffer.allocate(byteBuf.readableBytes());
+                byteBuf.readBytes(nioBuf);
+                readBuffer.writeBuffer(nioBuf);
```

Comment:
So here we make a temporary buffer which get bytes from netty one and added to the output stream as is without any copies?

Maybe it makes sense to add a comment saying this.
And maybe I would preserve a part of the removed comment in order to explain why we need to make a copy